src/main/java/co/cask/gcp/spanner/source/SpannerSource.java

```diff
@@ -133,17 +129,18 @@ public void prepareRun(BatchSourceContext batchSourceContext) throws Exception {
     configuration.set(SpannerConstants.SPANNER_BATCH_TRANSACTION_ID, getSerializedObjectString(batchTransactionId));
     configuration.set(SpannerConstants.PARTITIONS_LIST, getSerializedObjectString(partitions));
 
+    co.cask.hydrator.common.LineageRecorder lineageRecorder =
+      new co.cask.hydrator.common.LineageRecorder(batchSourceContext, config.referenceName);
+    lineageRecorder.createExternalDataset(config.getSchema());
```

Comment:
@albertshau This does not seem right. We write the JSON string to the topic so we should have the schema as just body string. Do you agree?

Reply:
Yes. Okay cool.